diff --git a/lib/NetConnection.js b/lib/NetConnection.js
index 3e4be6f259b004c39bce8aca091a167270df0a67..2716919a489146fe16db5cdeba7c5a910a90f95d 100644
--- a/lib/NetConnection.js
+++ b/lib/NetConnection.js
@@ -60,13 +60,15 @@ class NetConnection {
 			return false;
 		}
 		
-		const [, application, instance] = pathname.split('/');
+		const [, application] = pathname.split('/');
 		if (!application) {
 			return false;
 		}
 		
-		const app = `${application}${instance ? '/' : ''}${instance || ''}`;
-		const tcUrl = `${protocol}//${hostname}:${port}/${app}${search}`;
+		// app 只取第一段
+		const app = application;
+		// tcUrl 不包含查询参数（与 ffmpeg 行为一致）
+		const tcUrl = `${protocol}//${hostname}:${port}/${app}`;
 		
 		const client = new Client(hostname, port);
 		this[CLIENT] = client;
@@ -160,8 +162,9 @@ class NetConnection {
 		try {
 			const [, response] = await client.invoke('connect', null, {
 				app,
+				'type': 'nonprivate',
 				tcUrl,
-				'flashVer': 'WIN 32,0,0,465',
+				'flashVer': 'FMLE/3.0 (compatible; FMSc/1.0)',
 				'fpad': false,
 				'capabilities': 15,
 				'audioCodecs': 3191,
@@ -172,6 +175,7 @@ class NetConnection {
 			
 			this.isConnected = true;
 			this.uri = uri;
+			// 启用 PingPong 保活
 			this[PING_PONG].start(client);
 			
 			for (const name of [...new Set(this[SHAREDOBJECT])]) {
